lms/djangoapps/instructor/views/api.py

```diff
@@ -221,6 +223,25 @@ def wrapped(*args, **kwargs):  # pylint: disable=missing-docstring
     return decorator
 
 
+def require_sales_admin(func):
+    """
+    Decorator for checking sales administrator access before executing an HTTP endpoint. This decorator
+    is designed to be used for a request based action on a course. It assumes that there will be a
+    request object as well as a course_id attribute to leverage to check course level privileges.
+
+    If the user does not have privileges for this operation, this will return HttpResponseForbidden (403).
+    """
+    def wrapped(request, course_id):  # pylint: disable=missing-docstring
+        course_key = CourseKey.from_string(course_id)
```

Comment:
What happens if the course key isn't valid?  I think right now this will return a 500, but maybe a 404 would be more appropriate.